beacon-chain/state/stategen/service.go

```diff
@@ -96,6 +127,9 @@ func (s *State) Resume(ctx context.Context, fState state.BeaconState) (state.Bea
 	fRoot := bytesutil.ToBytes32(c.Root)
 	// Resume as genesis state if last finalized root is zero hashes.
 	if fRoot == params.BeaconConfig().ZeroHash {
+		if s.MinimumSlot() > 0 {
+			return nil, fmt.Errorf("no finalized checkpoint, and history before slot %d not currently available", s.MinimumSlot())
```

Comment:
I understand history before slot will not be available, but I don understand this check. Genesis state should be available

Reply:
I can change this PR to require genesis state, do you want me to do that? I can also make that change in the PR where we add the ability to kick off checkpoint sync[1] (otherwise these code paths will just never enter the if condition, since there will always be a genesis state at slot 0).

[1] https://github.com/prysmaticlabs/prysm/pull/10386

Reply:
I think we should require a genesis state eventually. If you want to do this PR, that's great, or you can do it later.
My only request is we don't forget to remove this dead code when we require a genesis state because that's easy to forget! so either does it now or create an issue and insert the todo with issue # here

Reply:
I'm going to just do it today.